src/sessions.rs

```diff
@@ -29,6 +30,36 @@ pub(crate) fn get_sessions() -> Result<Vec<String>, io::ErrorKind> {
     }
 }
 
+pub(crate) fn get_sessions_sorted_by_creation_date() -> Result<Vec<String>, io::ErrorKind> {
+    match fs::read_dir(&*ZELLIJ_SOCK_DIR) {
+        Ok(files) => {
+            let mut sessions_with_creation_date: Vec<(String, SystemTime)> = Vec::new();
+            files.for_each(|file| {
+                let file = file.unwrap();
+                let file_name = file.file_name().into_string().unwrap();
+                let file_created_at = file.metadata().unwrap().created().unwrap();
```

Comment:
@a-kenji 
Thanks for your review :)

To introduce `?` operator here, it would be a problem that an error type of `file.matadata()` and `io::ErrorKind` are much different. However, they can be into one error type using [anyhow library](https://crates.io/crates/anyhow). Would it be acceptable to add it?

By bringing it, `get_sessions_sorted_by_creation_date` will be:

```rust
pub(crate) fn get_sessions_sorted_by_creation_date() -> anyhow::Result<Vec<String>> {
    match fs::read_dir(&*ZELLIJ_SOCK_DIR) {
        Ok(files) => {
            let mut sessions_with_creation_date: Vec<(String, SystemTime)> = Vec::new();
            for file in files {
                let file = file?;
                let file_name = file.file_name().into_string().unwrap();
                let file_created_at = file.metadata()?.created()?;
                if file.file_type()?.is_socket() && assert_socket(&file_name) {
                    sessions_with_creation_date.push((file_name, file_created_at));
                }
            }
            sessions_with_creation_date.sort_by_key(|x| x.1); // the oldest one will be the first

            let sessions = sessions_with_creation_date
                .iter()
                .map(|x| x.0.clone())
                .collect();
            Ok(sessions)
        }
        Err(err) => {
            if let io::ErrorKind::NotFound = err.kind() {
                Ok(Vec::with_capacity(0))
            } else {
                Err(err.into())
            }
        }
    }
}
```

Many `unwrap` disappeared except for [`OsString::into_string`](https://doc.rust-lang.org/std/ffi/struct.OsString.html#method.into_string) which actually does not have an error type.

To properly treat and propagate errors, I think almost all `unwrap` should be removed and replaced with `anyhow`. So, I suggest doing refactor all functions, and some situation needs to introduce [`thiserror`](https://crates.io/crates/thiserror) as well. Additionally, the main function also can return `anyhow::Result` as shown [here](https://docs.rs/anyhow/1.0.44/anyhow/#details).

Reply:
Oh, I did not realize the `metadata` error was so much different!
Yes, I think adding `anyhow` should be fine. 

For which situations do you think `thiserror` needs to be used?

Reply:
@a-kenji 

> Oh, I did not realize the `metadata` error was so much different!
> Yes, I think adding `anyhow` should be fine.

Thanks!

> For which situations do you think `thiserror` needs to be used?

`anyhow` requires an error type to implement `std::error::Error` trait. This is why I could not apply `?` operator to [`OsString::into_string`](https://doc.rust-lang.org/std/ffi/struct.OsString.html#method.into_string). We, eventually, come to want to create a custom error type, but many traits such as `fmt::Display`, `std::error::Error`, and `From<_>` need implementing.

Accordingly, by introducing `thiserror`, most implementation can be one enum declaration like the example below; for example, the [`ConfigError`](https://github.com/zellij-org/zellij/blob/b37b9e81cb10ca9ab9fef05f99386bea02905030/zellij-utils/src/input/config.rs#L43) can be much improved:

https://docs.rs/thiserror/1.0.30/thiserror/index.html#example

Besides, it would be good to organize scattered errors on Zellij.

Reply:
If you want, I would add `thiserror` for another refactor to improve [`ConfigError`](https://github.com/zellij-org/zellij/blob/b37b9e81cb10ca9ab9fef05f99386bea02905030/zellij-utils/src/input/config.rs#L43) 👍

Reply:
If you want, I can add `thiserror` for another refactor to improve [`ConfigError`](https://github.com/zellij-org/zellij/blob/b37b9e81cb10ca9ab9fef05f99386bea02905030/zellij-utils/src/input/config.rs#L43) 👍

Reply:
Thank you for that explanation,
if it makes custom errors easier, I am all for it.

I would gladly look at a pr that improves and makes the
error structure easier.

Reply:
Thank you so much. I will commit to it :)